New Race Map with Bots

drakeaxelrod opened the pull request:
This new smce mod includes a new race track with cars that drive
themselves. booster plates to increase the speed of smce during the
race. a minimap to keep track of the posisition of everyone. a clock
that keeps track of the time of the race and each cars finish time.

drakeaxelrod committed:
New Race Map with Bots

This new smce mod includes a new race track with cars that drive
themselves. booster plates to increase the speed of smce during the
race. a minimap to keep track of the posisition of everyone. a clock
that keeps track of the time of the race and each cars finish time.

klarasve commented:
Fully functional and works, just need to update one line in arduino file to make mqtt work and show up on website:
`vroom.ino line 79: mqtt.onMessage(+[](String& topic, String& message) `
(by adding + and &)

klarasve reviewed (changes requested):
Fully functional and works, just need to update one line in arduino file to make mqtt work and show up on website:
vroom.ino line 79: `mqtt.onMessage(+[](String& topic, String& message)`
(by adding + and &)
Also looks really, really good and introduces a lot more functionalities such as minimap and bot cars!

drakeaxelrod committed:
Add improvements to minimap and player car model.

Made the minimap much more intuitive and useful. replaced the smce
cosmetic car pieces with same car model as bots. added smoke on turning
to the smce car.

drakeaxelrod committed:
Merge branch 'master' into race-map-with-bots

drakeaxelrod committed:
Remove the godot asset files

This should free up a decent chunk of space on github.

VernitaJ commented on godot-resources/mods/RaceAgainstMod/src/environments/raceagainst/replacements/smoke.gd:
```diff
@@ -0,0 +1,22 @@
+func _process(delta):
+
+	if ray.is_colliding() or ray2.is_colliding() or ray3.is_colliding() :
+		$smoke/Smoke.emitting = true
```
I really like this added feature. Great for the gaming environment

VernitaJ reviewed (approved):
This implementation is working well. The countdown and placement display are great additions for the gaming aspect. Speed boost is great, it's increased the fun in the application massively!
Some small additions to be added to this part of the project including difficulty variations, but that will be separate to this PR. Great job!

klarasve reviewed (approved):
Fully functional and works.

Axel-J-L reviewed (approved):
Great job with this Drake, appreciate working with you on this!
